docs/kubernetes/operator/instrumenting-nodejs.md

```diff
@@ -0,0 +1,94 @@
+(not user friendly, but we could use it in the future if we want to add a proper example here)
+-->
+
+In this section you will learn how to:
+
+- Enable auto-instrumentation of a Node.js application following any of the supported methods, such as:
+  - Adding an annotation to the deployment pods.
+  - Adding an annotation to the namespace.
+- Verify that auto-instrumentation libraries are injected and configured correctly.
+- Confirm data is flowing to **Kibana Observability**.
+
+For demonstration purposes, we assume the application to be instrumented is a deployment named `nodejs-app` running in the `nodejs-ns` namespace.
+
+1. Ensure you have successfully [installed the OpenTelemetry Operator](./README.md), and confirm that the following `Instrumentation` object exists in the system:
+
+```bash
+$ kubectl get instrumentation -n opentelemetry-operator-system
+NAME                      AGE    ENDPOINT                                                                                                
+elastic-instrumentation   107s   http://opentelemetry-kube-stack-daemon-collector.opentelemetry-operator-system.svc.cluster.local:4318
+```
+> [!NOTE]
+> If your `Instrumentation` object has a different name or is created in a different namespace, you will have to adapt the annotation value in the next step.
+
+2. Enable auto-instrumentation of the Node.js application using one of the following methods:
+
+  - Edit the workload definition and include the annotation under `spec.template.metadata.annotations`:
+
+    ```yaml
+    spec:
+    ...
+      template:
+        metadata:
+          labels:
+            app: nodejs-app
+          annotations:
+            instrumentation.opentelemetry.io/inject-nodejs: opentelemetry-operator-system/elastic-instrumentation
+    ...
+    ```
+
+  - Alternatively, add the annotation at namespace level to apply auto-instrumentation in all Pods of the namespace:
+
+    ```bash
+    kubectl annotate namespace nodejs-ns instrumentation.opentelemetry.io/inject-nodejs=opentelemetry-operator-system/elastic-instrumentation
+    ```
+
+3. Restart application:
+
+  Once the annotation has been set, the Pods need to be recreated for the instrumentation libraries to be injected.
+
+    ```bash
+    kubectl rollout restart deployment nodejs-app -n nodejs
+    ```
+
+4. Verify the auto-instrumentation resources are injected in the Pod:
+
+  Node.js apps are instrumented by the OpenTelemetry Operator with the following actions:
+(TBD / WE NEED TO REVIEW HOW THIS IS DONE / NODE_OPTIONS env var?)
+
+  - ??It adds an init container in the Pod with the objective of copying the SDK to a shared volume.
+
+  - ??Defines an `emptyDir volume` mounted in both containers.
+
+  - ??Adds `NODE_OPTIONS` and other OTEL related environment variables.
```

Comment:
The mechanism is basically the same as for Python and Java. I'd describe it in this order (slightly diff order to how you ahve it for the Python and Java docs):

1. An `emptyDir` volume (https://kubernetes.io/docs/concepts/storage/volumes/#emptydir) is created, named `opentelemetry-auto-instrumentation-nodejs` and is mounted in the app container and a new init container.
2. An init container, named `opentelemetry-auto-instrumentation-nodejs`, is created using the EDOT Node.js image `docker.elastic.co/observability/elastic-otel-node:<version>`. This init container copies the EDOT Node.js SDK to the shared volume.
3. Environment variables are added to the app container to configure (`OTEL_*` variables) and start (`NODE_OPTIONS` variable) the EDOT Node.js SDK.

It would be good to go through the example and get the output of the `kubectl describe pod ...` commands to show the values for Node.js. Let me know if you'd like me to do that to get the values.

Reply:
Now the description of the instrumentation steps has been moved to the `instrumenting_applications.md`, under `How auto-instrumentation works` section.

I have changed the order of the steps there, I agree it's better to start with the `emptyDir` volume.
You will see the change in the next commit, and in the `Node.js` specific doc we will have:

> 4. Verify the [auto-instrumentation resources](./instrumenting-applications.md#how-auto-instrumentation-works) are injected in the Pod: